src/run.ts

```diff
@@ -123,31 +103,21 @@ async function runLint(): Promise<null|Error> {
         return null;
     }
 
-    const pullRequestNumber = github.context.payload.pull_request?.number;
-    if (pullRequestNumber !== undefined) {
-        // If this action was configured for pull requests, we post the
-        // FileAnnotations as comments.
-        try {
-            await postComments(
-                authenticationToken,
-                owner,
-                repository,
-                pullRequestNumber,
-                result.fileAnnotations,
-            );
-        } catch (error) {
-            // Log the error, but continue so that we still write
-            // out the raw output to the user.
-            if (isError(error)) {
-                core.info(`Failed to write comments in-line: ${error.message}`);
-            } else {
-                core.info(`Failed to write comments in-line`);
-            }
+    // If this action was configured for pull requests, we post the
+    // FileAnnotations as comments.
+    result.fileAnnotations.forEach((fileAnnotation: FileAnnotation) => {
+        const { path, start_line, start_column, message } = fileAnnotation;
+        if (path === undefined || start_line === undefined || start_column === undefined) {
+          core.error(message);
+          return;
         }
-    }
-
-    // Include the raw output so that the console includes sufficient context.
-    return {
-        message: `buf found ${result.fileAnnotations.length} lint failures.\n${result.raw}`
-    };
+        // This uses the `::error` message feature of Github Actions. It converts the message to
+        // an error log in the Github Actions console and creates an error annotation at the given
+        // file path and line. This is not currently supported with `core.error`.
+        // For more information, see the documentation:
+        // https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message
+        core.info(`::error file=${path},line=${start_line},col=${start_column}::${message}`);
+    })
+    // All file annotations in the result have been processed, return an empty error.
+    return {message: ""};
```

Comment:
Everything looks great, but consider including a more meaningful message rather than an empty error.
```
    return {
        message: `buf found ${result.fileAnnotations.length} lint failures.\n
    };
```